store/database.go

```diff
@@ -0,0 +1,45 @@
+package store
```

Comment:
lets put these interfaces in core so users dont need to import store to use them

Reply:
why do you not want users to import store? This doesn't seem like idiomatic go to me personally.

Reply:
I guess the question is what else would be in store v2 besides interfaces?  In my mind store v2 was the default implementation of interfaces we defined for store in core.

Reply:
_implementors_ of these interfaces won't need to import store to do so (unless we include bespoke store types in the interface specification) since golang only requires that a type _satisfy_ (not reference) the interface contract. 

Of course we often write
```
	_ Tree = (*immutableTree)(nil)
```
referencing the interface anyway, but it's not required, and (by the way) creates tight coupling where there was only loose coupling.

The question for me is whether we want _consumers_ (e.g. keepers) to depend on store as the interface spec.  If consumers hold a reference to the implementing type through the interface described in store that would form a hard dependency and could make the design less modular and harder to replace store with a different implementation.

Reply:
For me it's less so consumers but if an app dev wants to implement a different store they now have to depend on store v2 for the Interfaxes potentially bringing in unneeded bloat. Putting interfaces into core is what we have been doing for the year for many thing why should this be any different.

Reply:
This is pretty backwards to me from an idiomatic domains separation POV, as in I'd expect implementations of `Animal` to exist in the `animal` package along side the `Animal` interface. 

E.g. Take the `rand.Source` interface defined by Golang stdlib https://pkg.go.dev/math/rand#Source. You don't see `Source` implemented in some "core" like package and the implementations of it (e.g. `Source64`) in `rand`.

In any case, not a hill I wanna fight on 😆. When the dust settles on the interfaces, I'll move them to core 👍